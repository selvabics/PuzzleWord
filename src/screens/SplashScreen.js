/**
 * This is Splash screen
 * Here we can implement any loading tasks such as calling API, set us user account etc
 */
import React, { useEffect } from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

import BackgroundWallper from '../components/backgroundWallpaper';
import { replace } from '../router/navigation';
import CommonStyles from '../utils/styles';

const SplashScreen = () => {

    useEffect(() => {
        // We can do any loading tasks here before we go into the app
        setTimeout(() => {
            replace('loginScreen');
        }, 500);
    }, []);

    return (
        <BackgroundWallper>
            <View style={[CommonStyles.flex, CommonStyles.center]}>
                <ActivityIndicator
                    style={styles.logo}
                    animating={true}
                    size={'small'}
                    color={'white'}
                />
            </View>
        </BackgroundWallper>
    );
};

const styles = StyleSheet.create({
});

export default SplashScreen;
