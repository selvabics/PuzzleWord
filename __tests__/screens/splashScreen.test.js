import renderer from 'react-test-renderer';
import SplashScreen from '../../src/screens/SplashScreen';

describe('Testing splach screen', () => {
    it('renders Splash Screen', () => {
        const tree = renderer
            .create(<SplashScreen />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

