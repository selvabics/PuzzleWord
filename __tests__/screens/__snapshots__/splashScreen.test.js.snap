// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing splach screen renders Splash Screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      {
        "flex": 1,
        "paddingBottom": 20,
        "paddingTop": 30,
      },
      {},
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/background.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="white"
      size="small"
    />
  </View>
</View>
`;
